app.py

```diff
@@ -115,119 +123,32 @@ def get_frame_range_from_shotgun(self):
 
         return ( data[sg_in_field], data[sg_out_field] )
 
-
     def get_current_frame_range(self, engine):
-
-        if engine == "tk-maya":
-            import pymel.core as pm
-            import maya.cmds as cmds
-            current_in = cmds.playbackOptions(query=True, minTime=True)
-            current_out = cmds.playbackOptions(query=True, maxTime=True)
-
-        elif engine == "tk-nuke" and not self.engine.hiero_enabled:
-            import nuke
-            current_in = int(nuke.root()["first_frame"].value())
-            current_out = int(nuke.root()["last_frame"].value())
-
-        elif engine == "tk-motionbuilder":
-            from pyfbsdk import FBPlayerControl, FBTime
-
-            lPlayer = FBPlayerControl()
-            current_in = lPlayer.LoopStart.GetFrame()
-            current_out = lPlayer.LoopStop.GetFrame()
-
-        elif engine == "tk-softimage":
-            import win32com
-            xsi = win32com.client.Dispatch('XSI.Application')
-
-            current_in = xsi.GetValue("PlayControl.In")
-            current_out = xsi.GetValue("PlayControl.Out")
-
-        elif engine == "tk-houdini":
-            import hou
-            current_in, current_out = hou.playbar.playbackRange()
-
-        elif engine == "tk-3dsmax":
-            from Py3dsMax import mxs
-            current_in = mxs.animationRange.start
-            current_out = mxs.animationRange.end
-        elif engine == "tk-3dsmaxplus":
-            import MaxPlus
-            ticks = MaxPlus.Core.EvalMAXScript("ticksperframe").GetInt()
-            current_in = MaxPlus.Animation.GetAnimRange().Start() / ticks
-            current_out = MaxPlus.Animation.GetAnimRange().End() / ticks
-
-        else:
-            raise tank.TankError("Don't know how to get current frame range for engine %s!" % engine)
-
-        return (current_in, current_out)
+        try:
+            result = self.execute_hook("hook_frame_operation",
```

Comment:
Ah one other thing we've spotted is that it's using the old method for hook execution.
The new method should be [execute_hook_method](https://developer.shotgunsoftware.com/tk-core/platform.html?highlight=execute_hook#sgtk.platform.Application.execute_hook_method).

I appreciate you submitted this sometime ago, if you like I can make these changes and continue to push it though?

Reply:
I've just noticed this. Of course, feel free to take it as far as you guys think it needs.  Thanks!

Reply:
In fact @Ahuge as in jumped in on it, and made the changes, which is awesome, so I'm hoping to push this through QA soon.
Thanks a lot both of you!